external-crates/move/move-abstract-stack/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module defines an abstract stack used by various verifier passes
+
+#[cfg(test)]
+mod unit_tests;
+
+use std::cmp::Ordering;
+use std::fmt::{self, Debug};
+use std::num::NonZeroU64;
+
+#[derive(Default, Debug)]
+/// An abstract value that compresses runs of the same value to reduce space usage
+pub struct AbsStack<T> {
```

Comment:
Thoughts on calling this `AbstractStack`? Or maybe `RLEStack`? `AbsStack` feels like a needless abbreviation to me/a bit jarring when reading the code.